packages/hub/schema/group.js

```diff
@@ -1,44 +1,29 @@
-const { get, isEqual, intersection } = require('lodash');
+const {
+  get,
+  isEqual,
+  intersection
+} = require('lodash');
 
 module.exports = class Group {
   constructor(model, allFields) {
-    let searchQuery = get(model, 'attributes.search-query');
-    if (!searchQuery) {
-      throw new Error(`group must have a search-query attribute: ${JSON.stringify(model, null, 2)}`);
+    if (model.type === 'groups' && model.id === 'everyone') {
+      // Skip validation for the `everyone` group. All users get assigned this group.
+      // See also bootstrap-schema.js and everyone-group.js for special handling.
+      this.id = model.id;
+      this.types = []; // required for downstream typechecks
+    } else {
+      let searchQuery = get(model, 'attributes.search-query');
+      this.types = validatedTypes(searchQuery, model);
+      this._fieldFilters = validatedFieldFilters(allFields,  searchQuery, model);
+      this._allFields = allFields;
+      this.id = model.id;
     }
-    let types = get(searchQuery, 'filter.type.exact');
```

Comment:
This big section got moved, unchanged, into two smaller functions called `validatedTypes` and `validatedFieldFilters`